src/model/transaction/moveBlockInContentState.js

```diff
@@ -13,83 +13,272 @@
 
 'use strict';
 
+import type {BlockNodeConfig} from 'BlockNode';
 import type {BlockNodeRecord} from 'BlockNodeRecord';
 import type ContentState from 'ContentState';
 import type {DraftInsertionType} from 'DraftInsertionType';
+import type {BlockMap} from 'BlockMap';
 
+const ContentBlockNode = require('ContentBlockNode');
 const invariant = require('invariant');
 
-function moveBlockInContentState(
+const getNextDelimiterBlockKey = (
+  blockToBeMoved: BlockNodeRecord,
+  blockMap: BlockMap,
+): ?string => {
+  const nextSiblingKey = blockToBeMoved.getNextSiblingKey();
+
+  if (nextSiblingKey) {
+    return nextSiblingKey;
+  }
+
+  const parent = blockToBeMoved.getParentKey();
+
+  if (!parent) {
+    return null;
+  }
+
+  let nextNonDescendantBlock = blockMap.get(parent);
+  while (
+    nextNonDescendantBlock &&
+    (!nextNonDescendantBlock.getNextSiblingKey() ||
+      !nextNonDescendantBlock.getParentKey())
+  ) {
+    nextNonDescendantBlock = blockMap.get(
+      nextNonDescendantBlock.getParentKey(),
+    );
+  }
+
+  if (!nextNonDescendantBlock) {
+    return null;
+  }
+
+  return nextNonDescendantBlock.getNextSiblingKey();
+};
+
+const updateBlockMapLinks = (
+  blockMap: BlockMap,
+  originalBlockToBeMoved: BlockNodeRecord,
+  originalTargetBlock: BlockNodeRecord,
+  insertionMode: DraftInsertionType,
+  isExperimentalTreeBlock: boolean,
+): BlockMap => {
+  if (!isExperimentalTreeBlock) {
+    return blockMap;
+  }
+  // insertionMode can only either be after or before
```

Comment:
I think this would be more clear if we say
```
// insertionMode can only either be 'after' or 'before'
```
At first I misunderstood and thought that it could only fall after or before some unnamed thing, but we mean that the possible values of the 'insertionMode' variable are 'after' and 'before'.
Maybe even something like this:
```
// possible values of 'insertionMode' are: 'after', 'before'
```